src/extension-runners/chromium.js

```diff
@@ -127,8 +127,58 @@ export class ChromiumExtensionRunner {
       chromeFlags.push(...this.params.args);
     }
 
-    if (this.params.chromiumProfile) {
-      chromeFlags.push(`--user-data-dir=${this.params.chromiumProfile}`);
+    function isUserDataDir(dirCandidate: string): boolean {
+      const localStateCandidate = path.join(dirCandidate, 'Local State');
+      const defaultCandidate = path.join(dirCandidate, 'Default');
+      // Local State and Default are typical for the user-data-dir
+      return fs.pathExistsSync(localStateCandidate)
+        && fs.pathExistsSync(defaultCandidate);
+
+    }
+
+    function isProfileDir(dirCandidate: string): boolean {
+      const securePreferencesCandidate = path.join(
+        dirCandidate,
+        'Secure Preferences');
+      //Secure Preferences is typical for a profile dir
+      return fs.pathExistsSync(securePreferencesCandidate);
+    }
+
+    let userDataDir = false;
+    const chromiumProfile = this.params.chromiumProfile;
+    if (chromiumProfile) {
+      const isProfileDirAndNotUserData =
+        fs.pathExistsSync(chromiumProfile)
```

Comment:
This part of the check is not needed. `isProfileDir` returning true implies that the directory exists.

Reply:
yes indeed